Treq and cryptography instead of txrequests and pycrypto

shyba opened the pull request:
Changes:
* change pycrypto to cryptography (we were mixing them)
* change txrequests to treq (avoids some thread pools)
* change requests to treq on Exchange Manager

Optional changes included here (easy to drop):
* change wallet download to support resuming downloads from s3 (download the missing part of the header file instead of everything)

Probably to change (maybe in another PR):
* add a method to check headers integrity on start up. This way we can automate the remove and download, avoiding the user to get frustrated solving it.

jackrobison commented on lbrynet/pointtraderclient/pointtraderclient.py:
```diff
@@ -1,16 +1,40 @@
+from cryptography.hazmat.backends import default_backend
```
The `pointtrader` stuff can be removed

jackrobison commented on lbrynet/tests/mocks.py:
```diff
@@ -15,6 +18,12 @@
 from util import debug_kademlia_packet
 
 KB = 2**10
+PUBLIC_EXPOENT = 65537  # http://www.daemonology.net/blog/2009-06-11-cryptographic-right-answers.html
```
typo - `PUBLIC_EXPONENT`

jackrobison commented on lbrynet/core/Wallet.py:
```diff
@@ -118,27 +93,39 @@ def __init__(self, storage):
-                        headers_file.write(raw_headers)
-                    log.info("fetched headers from s3 (s3 height: %i)", s3_height)
+        local_header_size = self.local_header_file_size()
+        resume_header = {"Range": "bytes={}-".format(local_header_size)}
```
Neat

shyba commented:
(hopefully) Fixed comments
Also added an usability workaround for checking headers file integrity and either:
1. delete headers file if the checksum mismatches (corrupted)
2. truncate to checkpoint if checksum matches but the size isn't divisible by the header size (incomplete write)
3. do nothing if it's divisible and checksum matches

Added this and the partial downloads support to the changelog.

shyba commented on lbrynet/core/Wallet.py:
```diff
@@ -118,27 +93,39 @@ def __init__(self, storage):
+                log.info("fetched headers from s3 (s3 height: %i)", s3_height)
             else:
-                log.error("invalid size for headers from s3")
+                log.warning("s3 is more out of date than we are")
```
changed

shyba commented:
@jackrobison added the integrity checks also after s3 download as discussed

tzarebczan commented:
@shyba what's the next step here? Does it need to be reviewed again? You'd need to remove Jack and add him back in as reviewer.

shyba commented:
@tzarebczan I think it's done. The last change was something we discussed some days ago after it was approved. I will reassign the review so we can track that.
